envi/archs/h8/emu.py

```diff
@@ -0,0 +1,1314 @@
+"""
+The H8 Emulator module.
+"""
+
+import sys
+import struct
+
+import emu
+import envi
+import envi.bits as e_bits
+from envi.archs.h8 import H8Module
+from envi.archs.h8.regs import *
+from disasm import H8RegDirOper
+
+# calling conventions
+class H8Call(envi.CallingConvention):
+    """
+    Implement calling conventions for your arch.
+    """
+    def execCallReturn(self, emu, value, ccinfo=None):
+        sp = emu.getRegister(REG_SP)
+        pc = struct.unpack(">H", emu.readMemory(sp, 2))[0]
+        sp += 2 # For the saved pc
+        sp += (2 * argc) # Cleanup saved args
+
+        emu.setRegister(REG_SP, sp)
+        emu.setRegister(REG_R0, value)
+        emu.setProgramCounter(pc)
+
+    def getCallArgs(self, emu, count):
+        return emu.getRegisters(0xf)  # r0-r3 are used to hand in parameters.  additional ph8s are stored and pointed to by r0
```

Comment:
Is this something special?  I'm not aware of any instance where getRegisters takes an arg...